libraries/architecture/src/main/kotlin/io/element/android/libraries/architecture/coverage/ExcludeFromJacocoGeneratedReport.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.architecture.coverage
+
+@Retention(AnnotationRetention.RUNTIME)
+@Target(
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.ANNOTATION_CLASS,
+)
+annotation class ExcludeFromJacocoGeneratedReport
```

Comment:
According to https://github.com/jacoco/jacoco/wiki/FilteringOptions#annotation-based-filtering, this annotation should meet the requirements to make the targets excluded form Jacoco report and so Kover.